src/components/Details.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react'
+import { useParams } from 'react-router-dom'
+import {PostType} from '../types/Post.types'
+import {Link} from "react-router-dom"
+import { getOne, getList } from '../api/pokemons.api'
+import { useState, useEffect } from 'react'
+import Pokemon from '../components/Pokemon'
+
+interface Pokemon {
+    id: string;
+    name: string;
+    ability: string;
+}
+
+let getDetail = async function(id: string): Promise<Pokemon> {
+    const ability = (await getOne(id)).data.abilities[0].ability.name;
+    const numId = parseInt(id);
+    const name = (await getList(numId - 1, 1)).data.results[0].name;
```

Comment:
No need to make second request for one pokemon from the list, one can obtain `name` from the `getOne` request too.

Reply:
![image](https://user-images.githubusercontent.com/15846431/201931356-706590c9-fa77-4cdc-9047-f44bcba89ace.png)